modules/unicornBidAdapter.js

```diff
@@ -0,0 +1,144 @@
+const UNICORN_PB_COOKIE_KEY = '__pb_unicorn_aud';
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['uncn'],
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid: bid => isBidRequestValid(bid),
+  buildRequests: (validBidRequests, bidderRequest) =>
+    buildRequests(validBidRequests, bidderRequest),
+  interpretResponse: (serverResponse, request) =>
+    interpretResponse(serverResponse, request)
+};
+
+/**
+ * Placement ID and Account ID are required.
+ * @param {BidRequest} bidRequest
+ * @returns {boolean}
+ */
+const isBidRequestValid = bidRequest => {
+  return !!bidRequest.adUnitCode && !!bidRequest.params.accountId;
+};
+
+/**
+ * @param {Array<BidRequest>} validBidRequests
+ * @param {any} bidderRequest
+ * @returns {ServerRequest}
+ */
+export const buildRequests = (validBidRequests, bidderRequest) => {
+  return {
+    method: 'POST',
+    url: UNICORN_ENDPOINT,
+    data: buildOpenRtbBidRequestPayload(validBidRequests, bidderRequest)
+  };
+};
+
+/**
+ * Transform BidRequest to OpenRTB-formatted BidRequest Object
+ * @param {Array<BidRequest>} validBidRequests
+ * @param {any} bidderRequest
+ * @returns {string}
+ */
+function buildOpenRtbBidRequestPayload(validBidRequests, bidderRequest) {
+  utils.logInfo(
+    '[UNICORN] buildOpenRtbBidRequestPayload.validBidRequests:',
+    validBidRequests
+  );
+  utils.logInfo(
+    '[UNICORN] buildOpenRtbBidRequestPayload.bidderRequest:',
+    bidderRequest
+  );
+  const imp = validBidRequests.map(br => {
+    const sizes = utils.parseSizesInput(br.sizes)[0];
+    return {
+      id: br.bidId,
+      banner: {
+        w: sizes.split('x')[0],
+        h: sizes.split('x')[1]
+      },
+      tagid: utils.deepAccess(br, 'params.placementId') || br.adUnitCode,
+      secure: 1,
+      bidfloor: parseFloat(utils.deepAccess(br, 'params.bidfloorCpm') || 0)
+    };
+  });
+  const request = {
+    id: bidderRequest.auctionId,
+    at: 1,
+    imp,
+    cur: UNICORN_DEFAULT_CURRENCY,
+    site: {
+      id: window.location.hostname,
+      publisher: {
+        id: utils.deepAccess(validBidRequests[0], 'params.accountId')
+      },
+      domain: window.location.hostname,
+      page: window.location.href,
+      ref: bidderRequest.refererInfo.referer
+    },
+    device: {
+      language: navigator.language,
+      ua: navigator.userAgent
+    },
+    user: {
+      id: getUid()
+    },
+    bcat: utils.deepAccess(validBidRequests[0], 'params.bcat') || [],
+    source: {
+      ext: {
+        stype: 'prebid_uncn',
+        bidder: BIDDER_CODE
+      }
+    }
+  };
+  utils.logInfo('[UNICORN] OpenRTB Formatted Request:', request);
+  return JSON.stringify(request);
+}
+
+const interpretResponse = (serverResponse, request) => {
+  utils.logInfo('[UNICORN] interpretResponse.serverResponse:', serverResponse);
+  utils.logInfo('[UNICORN] interpretResponse.request:', request);
+  const res = serverResponse.body;
+  var bids = []
+  res.seatbid.forEach(sb => {
```

Comment:
during my tests, `res` was `""` (empty string) and this caused `res.seatbid` to fail. This must be fixed.